validator/lib/validateExtensions.js

```diff
@@ -0,0 +1,72 @@
+'use strict';
+const Cesium = require('cesium');
+
+const defaultValue = Cesium.defaultValue;
+const defined = Cesium.defined;
+
+module.exports = validateExtensions;
+
+const EMPTY_ARRAY = [];
+
+const requiredExtensions = ['3DTILES_content_gltf'];
+
+/**
+ * Check if a tileset's extensions are valid.
+ *
+ * @param {Object} options An object with the following properties:
+ * @param {Object} options.tileset The tileset JSON.
+ * @returns {String} An error message if validation fails, otherwise undefined.
+ */
+function validateExtensions(options) {
+    const tileset = options.tileset;
+    const extensions = tileset.extensions;
+    const extensionsUsed = defaultValue(tileset.extensionsUsed, EMPTY_ARRAY);
+    const extensionsRequired = defaultValue(tileset.extensionsRequired, EMPTY_ARRAY);
+
+    if (defined(extensions)) {
+        for (const extensionName in extensions) {
+            if (extensions.hasOwnProperty(extensionName)) {
+                const extension = extensions[extensionName];
+                if (!extensionsUsed.includes(extensionName)) {
+                    return `${extensionName} must be included in extensionsUsed`;
+                }
+                if (requiredExtensions.includes(extensionName) && !extensionsRequired.includes(extensionName)) {
+                    return `${extensionName} must be included in extensionsRequired`;
+                }
+
+                let message;
+                if (extensionName === '3DTILES_content_gltf') {
+                    message = validate3DTilesContentGltf(extension);
+                }
+                if (defined(message)) {
+                    return `Error in ${extensionName}: ${message}`;
+                }
+            }
+        }
+    }
+}
+
+function validate3DTilesContentGltf(extension) {
```

Comment:
Three's nothing specific to `3DTILES_content_gltf` here, this code should be moved above into the loop before we call `.includes()`.

Reply:
This code is checking the inner glTF `extensionUsed` and `extensionsRequired`

```json
"extensionsUsed": ["3DTILES_content_gltf"],
"extensionsRequired": ["3DTILES_content_gltf"],
"extensions": {
  "3DTILES_content_gltf": {
    "extensionsUsed": ["KHR_texture_transform"],
    "extensionsRequired": ["KHR_texture_transform"],
  }
}
```